sunpy/coordinates/transformations.py

```diff
@@ -174,17 +174,23 @@ def wrapped_func(*args, **kwargs):
 
 
 def _observers_are_equal(obs_1, obs_2):
-    # Note that this also lets pass the situation where both observers are None
-    if obs_1 == obs_2:
+    # Handle None cases
+    if obs_1 is None and obs_2 is None:
```

Comment:
In order to do the `is_equivalent_frame` check below, all the `None` cases have to be dealt with first.